yt_dlp/extractor/panopto.py

```diff
@@ -0,0 +1,427 @@
+            'average_rating': delivery.get('AverageRating'),
+            'chapters': self._extract_chapters(delivery) or None,
+            'uploader': delivery.get('OwnerDisplayName') or None,
+            'uploader_id': delivery.get('OwnerId'),
+            'description': delivery.get('SessionAbstract'),
+            'tags': traverse_obj(delivery, ('Tags', ..., 'Content')),
+            'channel_id': delivery.get('SessionGroupPublicID'),
+            'channel': traverse_obj(delivery, 'SessionGroupLongName', 'SessionGroupShortName', get_all=False)
+        }
+
+        # Podcast stream is usually the combined streams. We will prefer that by default.
+        for stream in delivery.get('PodcastStreams', []):
+            streams.append(self._extract_stream(video_id, stream, base_info))
+
+        if not streams or self._configuration_arg('get_multistreams'):
+            for stream in delivery.get('Streams', []):
+                streams.append(self._extract_stream(video_id, stream, base_info))
+        streams = list(filter(None, streams))
+
+        if not streams:
+            self.raise_no_formats('Did not find any streams')
+
+        if len(streams) == 1:
+            return {
+                **streams[0],
+                **base_info,
+            }
+        else:
+            return {
+                **base_info,
+                '_type': 'multi_video',
+                'entries': streams,
+            }
+
+
+class PanoptoPlaylistIE(PanoptoBaseIE):
+    _VALID_URL = PanoptoBaseIE.BASE_URL_RE + r'/Pages/(Viewer|Embed)\.aspx.*(?:\?|&)pid=(?P<id>[a-f0-9-]+)'
+    _TESTS = [
+        {
+            'url': 'https://howtovideos.hosted.panopto.com/Panopto/Pages/Viewer.aspx?pid=f3b39fcf-882f-4849-93d6-a9f401236d36&id=5fa74e93-3d87-4694-b60e-aaa4012214ed&advance=true',
+            'info_dict': {
+                'title': 'Featured Video Tutorials',
+                'id': 'f3b39fcf-882f-4849-93d6-a9f401236d36',
+                'description': '',
+            },
+            'playlist_mincount': 36
+        },
+        {
+            'url': 'https://utsa.hosted.panopto.com/Panopto/Pages/Viewer.aspx?pid=e2900555-3ad4-4bdb-854d-ad2401686190',
+            'info_dict': {
+                'title': 'Library Website Introduction Playlist',
+                'id': 'e2900555-3ad4-4bdb-854d-ad2401686190',
+                'description': 'md5:f958bca50a1cbda15fdc1e20d32b3ecb',
+            },
+            'playlist_mincount': 4
+        },
+
+    ]
+
+    def _entries(self, base_url, playlist_id, session_list_id):
+        session_list_info = self._call_api(
+            base_url, f'/Api/SessionLists/{session_list_id}?collections[0].maxCount=500&collections[0].name=ViewableItemsOnly', playlist_id)
+        items = session_list_info['Items']
+        if len(items) == 500:
+            self.report_warning(
+                'There are 500 items in this playlist. There may be more but we are unable to get them' + bug_reports_message(), only_once=True)
```

Comment:
why add bug_reports_message to this?

Reply:
I was thinking it would be worthy of report since there is no handling for it. But this message can probably be removed. If a user thinks there is more than 500 videos then they will probably report an issue anyways.

Also, looking at the site js code, there looks to be no support for pagination (likely the reason why they load 500 videos) :laughing: